language-plutus-core/src/Language/PlutusCore/Constant/Dynamic/Call.hs

```diff
@@ -22,11 +22,11 @@ dynamicCallAssign
     :: KnownType a
     => DynamicBuiltinName
     -> (a -> IO ())
-    -> (a -> IO ExBudget) -- TODO do we actually need IO here?
+    -> (a -> ExBudget) -- TODO do we actually need IO here?
```

Comment:
I think, yes. In order to estimate how much resources an `IO` function requires, you might need an `IO` action. I guess it'd be even better to just merge those two together and have a single `a -> IO ExBudget` for both doing the `IO` action and counting its cost.

Reply:
In that case, `git revert` the newest commit.

Reply:
Ah, no, I played with the code and I decided not to commit anything and let people decide if we want to do with `unsafeCoerce` or (fixable) merge conflicts.

Reply:
So here is your PR back.